### PR TITLE
only monkeys get the monkey namepick prompt

### LINKED ISSUE #22915: Staff of Change "name" prompt can stack
(WEB REPORT BY: shiftyrail REMOTE: 206.221.180.138:7777)
# Revision
837e3fdb37ef0e869a0dbb5b574ebb5004705e8e

# Description

See title, if you get SoC'd a few times in a row, the prompts stack and steal focus from the game.
This means you can't type something or use hotkeys while the 3 or so prompts are active.

### PR DESCRIPTION
**Pull request by SonixApache**
@brndd why

closes #22915

### CODE REVIEW

**Comment by boy2mantwicethefam** [comment]
ok so basicly

**Comment by brndd** [comment]
>`// A standardized proc for turning a mob into a monkey`
>turns mobs into things other than monkeys

ok, I guess

File: code/modules/mob/transform_procs.dm [hunk 1]
```diff
@@ -26,24 +26,25 @@
...
-				else
-					if(alert(new_mob,"Do you really want the name:\n[newname]?",,"Yes","No") == "Yes")
-						break
-			if(newname)
-				new_mob.name = new_mob.real_name = newname
+		if(namepick)
+			spawn(10)
+				var/newname
+				for(var/i = 1 to 3)
+					newname = reject_bad_name(stripped_input(new_mob,"You have been turned into a monkey! Pick a monkey name for your new monkey self.", "Name change [4-i] [0-i != 1 ? "tries":"try"] left",""),1,MAX_NAME_LEN)
```
Comment (brndd) [inline code/modules/mob/transform_procs.dm]: Might want to change the message here if this is used for things other than monkeys.

**Comment by ShiftyRail** [comment]
Maybe it closes some issues
Although I'll admit being able to name yourself as a catbeast to "Oof" was fun.

File: code/modules/mob/transform_procs.dm (hunk 1 above)
Comment (SonixApache) [reply code/modules/mob/transform_procs.dm]: fair

**Comment by brndd** [comment]
What about an optional parameter that lets you specify the message that's displayed?

**Comment by SonixApache** [comment]
WYCI tbh

**Comment by brndd** [comment]
https://github.com/SonixApache/apachestation13/pull/5

**Comment by SonixApache** [comment]
if it breaks it's on you but good job regardless

**Comment by brndd** [comment]
me write bad code? impossible

**Comment by jknpj** [comment]
>A standardized proc for turning a mob into a monkey

Only monkeys using it doesn't mean that it was created just for monkeys. It's a transform helper, mostly.

**Comment by brndd** [comment]
Going by the contents of the file I'd wager a guess it was created for monkeys and then other things started using it. That, or the comment is just bogus.

**Pull request merged**